File: articles/taxonomy-of-strategic-ideas/c/taxonomy_score.c
#include <stdio.h>

typedef struct {
    const char *name;
    double category, level, maturity, evidence, function, relationships, retrieval, governance, ethics;
} TaxonomyRecord;

double score(TaxonomyRecord t) {
    return 0.13*t.category + 0.11*t.level + 0.11*t.maturity + 0.12*t.evidence +
           0.13*t.function + 0.11*t.relationships + 0.13*t.retrieval +
           0.09*t.governance + 0.07*t.ethics;
}

int main(void) {
    TaxonomyRecord records[] = {
        {"Strategic Learning Repository", 0.78, 0.76, 0.72, 0.72, 0.80, 0.76, 0.82, 0.74, 0.66},
        {"Participatory Governance Prototype", 0.76, 0.74, 0.76, 0.70, 0.74, 0.72, 0.76, 0.70, 0.82},
        {"Advisory Panel Without Authority", 0.66, 0.62, 0.72, 0.62, 0.60, 0.66, 0.62, 0.56, 0.76}
    };
    int count = sizeof(records) / sizeof(records[0]);
    for (int i = 0; i < count; i++) {
        double value = score(records[i]);
        printf("%s | taxonomy strength %.3f | taxonomy risk %.3f\n", records[i].name, value, 1.0 - value);
    }
    return 0;
}

File: articles/taxonomy-of-strategic-ideas/rust/taxonomy_score.rs
// Command-line taxonomy scoring scaffold.
// Compile: rustc taxonomy_score.rs -o taxonomy_score
// Run: ./taxonomy_score

struct TaxonomyRecord {
    name: &'static str,
    category: f64,
    level: f64,
    maturity: f64,
    evidence: f64,
    function: f64,
    relationships: f64,
    retrieval: f64,
    governance: f64,
    ethics: f64,
}

fn score(t: &TaxonomyRecord) -> f64 {
    0.13 * t.category
        + 0.11 * t.level
        + 0.11 * t.maturity
        + 0.12 * t.evidence
        + 0.13 * t.function
        + 0.11 * t.relationships
        + 0.13 * t.retrieval
        + 0.09 * t.governance
        + 0.07 * t.ethics
}

fn main() {
    let records = vec![
        TaxonomyRecord { name: "Strategic Learning Repository", category: 0.78, level: 0.76, maturity: 0.72, evidence: 0.72, function: 0.80, relationships: 0.76, retrieval: 0.82, governance: 0.74, ethics: 0.66 },
        TaxonomyRecord { name: "Participatory Governance Prototype", category: 0.76, level: 0.74, maturity: 0.76, evidence: 0.70, function: 0.74, relationships: 0.72, retrieval: 0.76, governance: 0.70, ethics: 0.82 },
        TaxonomyRecord { name: "Advisory Panel Without Authority", category: 0.66, level: 0.62, maturity: 0.72, evidence: 0.62, function: 0.60, relationships: 0.66, retrieval: 0.62, governance: 0.56, ethics: 0.76 },
    ];

    for t in records {
        let value = score(&t);
        println!("{} | taxonomy strength {:.3} | taxonomy risk {:.3}", t.name, value, 1.0 - value);
    }
}
